files/usr/bin/cinnamon-xlet-makepot

```diff
@@ -0,0 +1,274 @@
+    parser.add_argument('-r', '--remove', action='store_true', dest='remove', default=False,
+        help=('The opposite of install, removes translations from the store. Again, it uses the UUID '
+              'to find the correct files to remove.'))
+    parser.add_argument('-k', '--keyword', dest='keyword', default='_',
+        help='Change the variable name gettext is assigned to in your files. The default is _.')
+    parser.add_argument('-o', '--potfile', dest='out_file', default=None,
+        help=('Use this option to specify the location for the generated .pot file. By default '
+              '<uuid>/po/<uuid>.pot is used. This is where translators are expecting to find the pot '
+              'file, so you should only use this option if you want the pot file elsewhere for your own use.'))
+    parser.add_argument('dir', help='the path to the applet/desklet/extension directory')
+
+    args = parser.parse_args()
+
+    dir = os.path.abspath(args.dir)
+    if not os.path.exists(dir):
+        print('%s does not exist' % dir)
+        quit()
+
+    uuid = os.path.basename(dir)
+
+    if os.path.exists(os.path.join(dir, 'files', uuid)):
+        dir = os.path.join(dir, 'files', uuid)
+
+    if args.install:
+        do_install(uuid, dir)
+        quit()
+
+    if args.remove:
+        do_remove(uuid)
+        quit()
+
+    if args.out_file is not None:
+        pot_path = os.path.abspath(args.out_file)
+    else:
+        pot_path = os.path.join(dir, 'po', uuid + '.pot')
+
+    pwd = os.getcwd()
+    os.chdir(dir)
+    if args.js or args.py:
+        try:
+            import subprocess
+            output = subprocess.check_output(["xgettext", "--version"])
+        except OSError:
+            print("xgettext not found, you may need to install the gettext package")
+            quit()
+
+    if args.js:
+        print("Scanning JavaScript files...")
+
+        js_files = []
+        for root, dirs, files in os.walk(dir):
+            rel_root = os.path.relpath(root)
+            js_files += [os.path.join(rel_root, file) for file in files if file.endswith('.js')]
+        js_files
+        if len(js_files) == 0:
+            print('none found')
+        else:
+            print('found %i file(s)\n' % len(js_files))
+        os.system("xgettext --language=JavaScript --keyword=%s --output=%s %s" % (
+            args.keyword,
+            pot_path,
+            ' '.join(js_files)
+        ))
+
+    if args.python:
+        print("Scanning for python files...")
+
+        py_files = []
+        for root, dirs, files in os.walk(dir):
+            rel_root = os.path.relpath(root)
+            py_files += [os.path.join(rel_root, file) for file in files if file.endswith('.py')]
+        if len(py_files) == 0:
+            print('none found')
+        else:
+            print('found %i file(s)\n' % len(py_files))
+        os.system("xgettext -j --language=JavaScript --keyword=%s --output=%s %s" % (
```

Comment:
Same as in the previous call to the `xgettext` command.

Reply:
Additionally, the `-j` (`--join-existing`) argument should be passed to the `xgettext` command **only** if the POT file already exists. If the scanning of the JS files didn't find any strings to translate (or if the `--skip-js` argument is used), a POT file will not be created and executing the `xgettext` command with the `--join-existing` argument with a non existent file for the `--output` argument will simply fail.